core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -505,15 +507,15 @@ public void updateTask(TaskResult result) throws Exception {
                 break;
         }
 
-        decide(workflowId);
-
         if (task.getStatus().isTerminal()) {
             long duration = getTaskDuration(0, task);
             long lastDuration = task.getEndTime() - task.getStartTime();
             Monitors.recordTaskExecutionTime(task.getTaskDefName(), duration, true, task.getStatus());
             Monitors.recordTaskExecutionTime(task.getTaskDefName(), lastDuration, false, task.getStatus());
         }
 
+        decide(workflowId);
```

Comment:
Is this moved here to fix the  recording of the metrics ?

Reply:
Yes, `decide(workflowId)` will be remove runtime data.